src/satellite_images_nso_datascience/model_metrics/custom_model_metrics.py

```diff
@@ -0,0 +1,296 @@
+        self.scaler = ascaler
+        self.location = alocation
+        self.satellite_constellation = asatellite_constellation
+
+        # Init model locations
+        if alocation == "Voornes Duin":
+            self.selected_features = [
+                "r",
+                "g",
+                "b",
+                "n",
+                "e",
+                "d",
+                "ndvi",
+                "re_ndvi",
+                "height",
+            ]
+
+        elif alocation == "Coepelduynen":
+            # Optimal parameters and features
+            self.selected_features = ["r", "g", "b", "n", "e", "d", "ndvi", "re_ndvi"]
+
+        elif alocation == "Schippersgat" and asatellite_constellation == "Superview":
+            # Optimal parameters and features
+            self.selected_features = ["r", "g", "b", "i", "ndvi", "height"]
+
+        elif alocation == "Schippersgat" and asatellite_constellation == "PNEO":
+            # Optimal parameters and features
+            self.selected_features = ["r", "g", "b", "n", "e", "d", "ndvi", "re_ndvi"]
+
+        elif alocation == "Nieuwkoopse_plassen" and asatellite_constellation == "PNEO":
+            # Optimal parameters and features
+            self.selected_features = ["r", "g", "b", "n", "e", "d", "ndvi", "re_ndvi"]
+
+        elif (
+            alocation == "Nieuwkoopse_plassen"
+            and asatellite_constellation == "Superview"
+        ):
+            # Optimal parameters and features
+            self.selected_features = ["r", "g", "b", "i", "ndvi", "ndwi"]
+
+    def metric_difficult_pixels(self):
+        """
+
+        Test the model performance on pixels which are edge cases and difficult to predict.
+
+        TODO: Added more of these difficult pixels.
+        """
+
+        fault_check = []
+
+        # Fill with more locations than only nieuwkoopse plassen, when this project goes on.
+        if (
+            self.location == "Nieuwkoopse_plassen"
+            and self.satellite_constellation == "Superview"
+        ):
+            # TODO: FIll with more difficult pixels.
+            fault_check.append(
+                [
+                    "Water",
+                    self.model.predict(
+                        self.scaler.transform([[275, 208, 117, 219, 97, 88]])
+                    )
+                    == "Water",
+                ]
+            )
+
+            fault_check.append(
+                [
+                    "Water",
+                    self.model.predict([[1024, 656, 575, 416, 122, 57]]) == "Water",
+                ]
+            )
+
+        if len(fault_check) == 0:
+            return (
+                "Error no difficult pixels found for the location or the constellation."
+            )
+
+        return fault_check
+
+    def metrics_on_small_tif_files(self):
+        """
+        Test the model on small random .tif files, one side for model performance on the other side for the connection with raw .tif files.
```

Comment:
I'm not sure what you mean with "on the other side for the connection with raw .tif files"?

Reply:
you could have testing this just on a normal dataframe but this functionality test if it also works on the raw .tif files.
It's more of a integration test

Reply:
All right.